File: src/main/scala/space/yangshuai/ojsolutions/nod/contest/contest24/SolutionA.scala
package space.yangshuai.ojsolutions.nod.contest.contest24

/**
  * Created by rotciv on 2017/4/28.
  */
object SolutionA {

  def main(args: Array[String]): Unit = {

    val n = scala.io.StdIn.readLine().toInt - 1
    val arr = new Array[Int](n)


    if ((1 + n) * 3 % 2 == 1) {
      println(0)
    } else {
      val total = (1 + n) * 3 / 2
      arr(0) = total - n - 1
      arr(1) = 1
      for (i <- 2 until n) {
        arr(i) = total - (arr(i - 1) + n + 1 - i)
      }
      println((n to 1 by -1).mkString(" "))
      println(arr.mkString(" "))
    }
  }

}

File: src/main/java/space/yangshuai/ojsolutions/leetcode/contest/contest135/SolutionA.java
package space.yangshuai.ojsolutions.leetcode.contest.contest135;

/**
 * @author yangshuai on 2019-05-05.
 */
public class SolutionA {

    class Point {
        float x;
        float y;
        Point(int[] arr) {
            this.x =arr[0];
            this.y = arr[1];
        }

        public boolean equals(Point other) {
            return x == other.x && y == other.y;
        }
    }

    public boolean isBoomerang(int[][] points) {

        Point a = new Point(points[0]);
        Point b = new Point(points[1]);
        Point c = new Point(points[2]);

        if (a.equals(b) || b.equals(c) || a.equals(c)) {
            return false;
        }

        if (a.y - b.y == 0 && b.y - c.y == 0) {
            return false;
        }

        if (a.y - b.y == 0 && b.y - c.y != 0) {
            return true;
        }

        if (a.y - b.y != 0 && b.y - c.y == 0) {
            return true;
        }



        return ((a.x - b.x) / (a.y - b.y)) != ((b.x - c.x) / (b.y - c.y));
    }

    public static void main(String[] args) {
        int[][] arr = new int[][]{{1, 0}, {0, 0}, {2, 0}};
        new SolutionA().isBoomerang(arr);
    }

}
